src/infra/src/table/folders.rs

```diff
@@ -139,8 +139,8 @@ async fn get_model(
 }
 
 /// Lists all folder ORM models with the specified type.
-async fn list_models(
-    db: &DatabaseConnection,
+async fn list_models<C: ConnectionTrait>(
+    db: &C,
```

Comment:
Swap the concrete `DatabaseConection` type for the `ConnectionType` trait so we can pass in a transaction.

This isn't strictly necessary for this PR, but seems like a good pattern to follow in general for anywhere where we need to pass a DB connection as a param